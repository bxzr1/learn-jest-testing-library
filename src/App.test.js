import { render, screen, waitFor } from '@testing-library/react';
import axios from 'axios';
import App from './App';
import renderer from 'react-test-renderer'; // used for snapshot testing

jest.mock('axios');

const fakeUsers = [{
    "id": 1,
    "name": "Test User 1",
    "username": "testuser1",
   }, {
    "id": 2,
    "name": "Test User 2",
    "username": "testuser2",
   }];

describe('App component', () => {
    test('it renders', async () => {
        // after setting up jest.mock('axios'), need to add the .get.mockResolvedValue line to old tests too
        axios.get.mockResolvedValue({data: fakeUsers});
        render(<App />);
        // only fails if thereâ€™s a compilation error or an error in the function component that impedes its rendering
        // though valid, it is not a complete test because it does not perform any assertions
        expect(screen.getByText('Users:')).toBeInTheDocument();

        // expect(screen.getByTestId('user-list')).toBeInTheDocument();
        // user-list is only created and populated after the fetch but the test is running before it completes, so turn this into an async-await operation

        const userList = await waitFor(() => screen.getByTestId('user-list'));
        expect(userList).toBeInTheDocument();
    });

    test('it displays a row for each user', async () => {
        axios.get.mockResolvedValue({data: fakeUsers});
        render(<App/>);

        const userList = await waitFor(() => screen.findAllByTestId('user-item'));
        expect(userList).toHaveLength(2);
    });
})

describe('App component snapshot', () => {
    test('it renders a correct snapshot', async () => {
        axios.get.mockResolvedValue({data: fakeUsers});
        const tree = renderer.create(<App/>).toJSON();
        expect(tree).toMatchSnapshot();
    })
})