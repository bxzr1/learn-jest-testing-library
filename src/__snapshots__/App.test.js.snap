// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component snapshot it renders a correct snapshot 1`] = `
<div
  className="App"
>
  <div>
    Users:
  </div>
  <div>
    Wait! Loading users...
  </div>
</div>
`;
